source/MaterialXRuntime/Private/PvtApi.cpp

```diff
@@ -121,6 +122,16 @@ void PvtApi::registerPrims(RtStagePtr stage)
         else if (prim.hasApi<RtNodeImpl>())
         {
             registerNodeImpl(prim);
+            RtNodeImpl nodeImpl(prim);
+            RtString nodeGraphName = nodeImpl.getNodeGraph();
+            if (nodeGraphName != RtString::EMPTY)
+            {
+                RtPrim nodeGraphPrim = stage->getPrimAtPath(RtPath(nodeGraphName.str()));
+                if (nodeGraphPrim)
+                {
+                    registerNodeGraph(nodeGraphPrim);
```

Comment:
This must be done before trying to make connections other the nodegraph is not "registered".